src/main/java/org/cactoos/http/io/BoundedByteBuffer.java

```diff
@@ -88,21 +92,27 @@ public void offer(final byte add) {
      */
     public boolean equalTo(final byte[] bytes) {
         final int result;
-        if (this.current < this.limit) {
-            result = new UncheckedScalar<>(
-                new Equality<Bytes>(
-                    new BytesOf(Arrays.copyOf(this.internal, this.current)),
-                    new BytesOf(bytes)
-                )
-            ).value();
+        result = new UncheckedScalar<>(
+            new Equality<Bytes>(
+                new BytesOf(currentState()),
+                new BytesOf(bytes)
+            )
+        ).value();
+        return result == 0;
+    }
+
+    private byte[] currentState(){
```

Comment:
@golyalpha this method seems a bit complex... In a way you moved some small complexity from the `offer` method to some big complexity in the `currentState` method. And you moved the array copy from one lace to another too.

I would prefer for the `equalTo` method to directly do a comparison in an imperative way on the two arrays following the algorithm described at the end of #60 instead of all this complexity. Like this we would have zero array copy :)

Reply:
Alright. I dropped the current state method, and now equalTo compares the array given to it with the internal buffer according to what you outlined in #60.
Not sure how exactly is that any less complex than copying arrays together to form what we want, because right now it's a bunch of for loops that can exit early if possible.

Reply:
@golyalpha I agree it's not pretty, but at least there are no useless array copy, which was the theme of the issue you are solving. It is maybe a good incentive for adding a `@todo` to make this more OO if you have some ideas of how to do that. Or maybe you can open a new issue to make this nicer actually, and we can discuss it there? This question is a good test for doing efficient OO code!